utils.groovy

```diff
@@ -482,6 +496,65 @@ def run_cloud_tests(pipecfg, stream, version, s3_stream_dir, basearch, commit) {
     parallel testruns
 }
 
+// Runs followup upgrade tests based on conditions
+def run_upgrade_tests(pipecfg, stream, version, basearch, commit) {
+    def stream_info = pipecfg.streams[stream]
+
+    def min_supported_start_versions = [
+        aarch64: 34,
+        ppc64le: 40, // We haven't released on ppc64le yet
+        s390x:   36,
+        x86_64:  32,
+    ]
+
+    // Because of an x86_64 bootloader issue the oldest that is supported
+    // on x86_64 is 32.x: https://github.com/coreos/fedora-coreos-tracker/issues/1448
+    // All other arches were shipped later than 32 so a min_version of 32 works.
+    def min_version = 32 as Integer
+    // Use a maximum version that matches the current stream Fedora major version
+    def manifest = readYaml(text:shwrapCapture("cosa shell -- cat src/config/manifest.yaml"))
+    def max_version = manifest.releasever as Integer
+
+    // A list of all the start versions we want to run tests for
+    def start_versions = []
+
+    // If this is a production build we'll run a test from the
+    // earliest release of each Fedora major we've supported for this
+    // stream. Iterate over the possibilities and kick off a job for
+    // each.
+    if (stream_info.type == 'production') {
+        for (start_version in min_version..max_version) {
+            start_versions += start_version
+        }
```

Comment:
yeah. I was thinking a bit more about this and I think it would be good to go ahead and run the full matrix for `testing` and `next` for the same reason as we want to run tests against `rawhide`. i.e. if there were a breaking change we'd want to know it when we were getting ready to release `testing` rather than later when we promote that content to stable.

I'm not super firm on this, but since we only run new builds on average every two weeks, it seemed like a little additional insurance for not a lot of drawback.

Reply:
SGTM, we can always adjust in the future too.